mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/spi/PojoMappingDelegate.java

```diff
@@ -30,6 +31,8 @@
 
 	FailureHandler failureHandler();
 
+	TimingSource timingSource();
```

Comment:
Wouldn't it be simpler to just expose the `TimeoutManager` to the mapper, and not the `TimingSource`? I thought that was the reason you moved the `TimeoutManager` to the engine... I'm a bit confused.

Reply:
In the current solution the engine can provide the timing source, this instance is shared with all the modules that need it. On the other hand the timeout manager is created each time a module needs it by the module itself. So that the module can control its lifecycle. Another benefit is the fact that a client module can choose its own subclass implementation, e.g.: `LuceneTimeoutManager`.